azure-devops/azext_devops/dev/team/invoke.py

```diff
@@ -0,0 +1,148 @@
+        service_list = []
+
+        for x in resource_areas:
+            if x.location_url not in service_list:
+                service_list.append(x.location_url)
+
+        resouce_locations = []
+
+        for x in service_list:
+            try:
+                logger.info('trying to get locations from ' + x)
+                clientMock =  VssClient(x, connection._creds)
+                resrouce_location_on_this_service = clientMock._get_resource_locations(all_host_types=True)
+                resouce_locations.extend(resrouce_location_on_this_service)
+            except:
+                logger.info('Failed to get location for ' + x)
+
+        return resouce_locations
+
+    client_url = ''
+    if not resource_areas:
+        #this is for on-prem
+        client_url = connection.base_url
+    
+    for resource_area in resource_areas:
+        if resource_area.name.lower() == area.lower():
+            client_url = resource_area.location_url
+
+    if not client_url:
+        raise CLIError('--area is not present in current organization')
+
+    client = VssClient(client_url, connection._creds)
+
+    # there can be multiple resouce/ area with different version so this version comparision is needed
+    location_id = ''
+    current_version = 0.0
+    resource_locations = client._get_resource_locations(all_host_types=True)
+    for resource_location in resource_locations:
+        if resource.lower() == resource_location.resource_name.lower() and area.lower() == resource_location.area.lower():
+            current_maxVersion = float(resource_location.max_version)
+            if current_maxVersion > current_version and  version >= current_version:
+                location_id = resource_location.id
+                current_version = current_maxVersion
+
+    if not location_id:
+        raise CLIError('--resource and api-version combination is not correct')
+
+    route_values = stringToDict(route_parameters)
+    query_values = stringToDict(query_parameters)
+
+    response = client._send(http_method=http_method,
+                            location_id=location_id,
+                            version=api_version,
+                            query_parameters=query_values,
+                            route_values=route_values,
+                            media_type=media_type,
+                            accept_media_type=accept_media_type,
+                            content=request_body)
+
+    logger.info('content type header')
+    logger.info(response.headers.get("content-type"))
+    if 'json' in response.headers.get("content-type") and not out_file:
+        return response.json()
+
+    if not out_file:
+        raise CLIError('response is not json, you need to provide --out-file where it can be written')
+
+    import os
+    if os.path.exists(out_file):
+        raise CLIError('out file already exists, please give a new name')
+
+    if not os.path.exists(out_file):
+        open(out_file, "a").close()
```

Comment:
For my info: Why do we need to close a handle here?

Reply:
this is like a touch to create file.. and we need to close handle so that we can take a new handle and write binary content later